GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs

```diff
@@ -763,6 +773,398 @@ private List<int> GetStepPosition(DataRow[] allSteps, DataRow stepData, List<int
+            
+            var requestMethod = (Method) questionApi.GetDetailValue<int>("request_type");
+            
+            var restClient = new RestClient();
+            var restRequest = new RestRequest(endpoint, requestMethod);
+            
+            // If a body is provided perform replacements and add it to the request.
+            if (!String.IsNullOrWhiteSpace(requestJson))
+            {
+                requestJson = await ReplaceConfiguratorItemsAsync(requestJson, configuration, false);
+                requestJson = await ReplaceConfiguratorItemsAsync(requestJson, configuration, false);
+                requestJson = await stringReplacementsService.DoAllReplacementsAsync(requestJson, extraData, removeUnknownVariables: false);
+
+                var regex = new Regex("([\"'])?{[^\\]}\\s]*}([\"'])?", RegexOptions.Compiled | RegexOptions.IgnoreCase, TimeSpan.FromMilliseconds(2000));
+                requestJson = regex.Replace(requestJson, "null");
+
+                restRequest.AddBody(requestJson, MediaTypeNames.Application.Json);
+            }
+            
+            await AddAuthenticationToApiCall(restRequest, questionApi);
+            await AddAcceptLanguageToApiCall(restRequest);
+            await AddCustomHeadersAsync(restRequest, questionApi, vueConfiguration: configuration, extraData: extraData);
+            
+            var restResponse = await DoExternalConfiguratorApiCallAsync(restClient, restRequest);
+            if (!restResponse.IsSuccessful || restResponse.Content == null)
+            {
+                logger.LogWarning("Error while trying to get the configuration from API ({questionApi}). The API response HTTP code was '{restResponseStatusCode}' and the result was: {restResponseContent}.\n\n{requestJson}", questionApi.Title, restResponse.StatusCode, restResponse.Content, requestJson);
+                stepData.Options = options;
+                return;
+            }
+            
+            // Get the question from the response.
+            JToken responseData = restResponse.Content.StartsWith("{") ? JObject.Parse(restResponse.Content) : JArray.Parse(restResponse.Content);
+            var keyParts = new List<string>(questionKey.Split('.'));
+            JToken currentObject = responseData;
+
+            // Step into the object till the correct object is found or the path leads to an invalid object.
+            while (keyParts.Count > 0)
+            {
+                if (keyParts[0].StartsWith("["))
+                {
+                    var selector = keyParts[0].Substring(1, keyParts[0].Length - 2);
+                    var selectorParts = selector.Split('=');
+
+                    var matchFound = false;
+
+                    foreach (JToken jToken in currentObject)
+                    {
+                        if (jToken.Type != JTokenType.Object || !jToken[selectorParts[0]].Value<string>().Equals(selectorParts[1], StringComparison.InvariantCultureIgnoreCase))
```

Comment:
Here you assume that selectorParts always contains at least 2 values. Are you 100% sure that is always the case? If not, please add a check for that.